packages/plugin-ext/src/main/browser/view/view-registry.ts

```diff
@@ -30,42 +36,51 @@ export class ViewRegistry {
     @inject(FrontendApplicationStateService)
     protected applicationStateService: FrontendApplicationStateService;
 
-    private containersWidgets: Map<string, ViewsContainerWidget> = new Map<string, ViewsContainerWidget>();
-    private promiseChain: Promise<void>;
+    private treeViewWidgets: Map<string, TreeViewWidget> = new Map();
+    private containerWidgets: Map<string, ViewsContainerWidget> = new Map();
+    private addPermissionPromise: Promise<void>;
 
-    registerViewContainer(location: string, viewContainer: ViewContainer) {
-        const widget = new ViewsContainerWidget(viewContainer);
-        this.containersWidgets.set(viewContainer.id, widget);
-        this.updatePromiseChain(() => {
-            if (widget && !this.applicationShell.getTabBarFor(widget)) {
-                this.applicationShell.addWidget(widget, {
-                    area: ApplicationShell.isSideArea(location) ? location : 'left'
-                });
-            }
-        });
+    @postConstruct()
+    init() {
+        this.addPermissionPromise = this.applicationStateService.reachedState(READY);
```

Comment:
could be renamed to `updateContainerViewsOnApplicationReady` ? maybe ?

Reply:
updateContainerPromise

Reply:
why `promise` ? promise is a type

Reply:
done

Reply:
```updateContainerOnApplicationReady```